### ISSUE TITLE
Questions about diffusion implementation code in trainer_brast.py

### ISSUE THREAD

**Issue by xiyuez2** [mobaidoctor/med-ddpm#14]
Thank you for your work and for sharing your code. This is the only diffusion model on 3D dense MRI images I can find!

I have two questions:
1. In the loss function in diffusion_model/trainer_brast.py (line 285), the hybrid loss is a hybrid of l2 loss and cross_entroy loss. May I ask why we use binary_cross_entropy_with_logits? Does this gives better results emprically? This loss assumes the ground truth is between 0 and 1, which is not the case since the GT is a gaussian distributed noise?

2. In the diffusion diffusion_model/trainer_brast.py (line205), we clamp the reconstructed x_t to be between -1 and 1. However, from the sample files provided, the data should be normalized between 0 and 1. Will the model perform better if we normalize the data to -1 and 1 or change the clamp function to clamp the data to 0-1?

Thanks in advance!

**Comment by mobaidoctor** [comment]
Thank you for your inquiry.

1. We apologize for not reverting the code to our original setup before releasing it in our repository. The hybrid loss was from an experiment in another study. We appreciate you bringing this issue to our attention. We will clean up our code. Please refer to our manuscript for a detailed description of our setup. For this study, we only used L1 loss.
2. We clamp the reconstructed x_t to the range of [-1, 1], because our training input images are normalized to this range, as can be verified from the samples provided in our repository under the directory at: "https://github.com/mobaidoctor/med-ddpm/tree/main/dataset/brats2021". The segmentation mask is in a different range [0, 1] after one-hot encoding is applied, because we disregard the background value of 0 and focus solely on the brain and tumor labels. You do not need to clamp the value to [0, 1] since our training images are in the range of [-1, 1] and our expected output is an image, not a segmentation mask.

We hope our responses are helpful for your study. If you have any further questions, please do not hesitate to contact us. As we are conducting research in the same domain, we would be delighted to assist and are open to collaboration. Good luck with your study.

**Comment by xiyuez2** [comment]
I see. Thanks!
